src/main/java/io/apicurio/datamodels/asyncapi/io/AaiDataModelWriter.java

```diff
@@ -390,8 +390,13 @@ public void visitOperation(Operation node) {
         Object parent = this.lookupParentJson(node);
         Object json = this.writeOperationBase(aaiNode);
 
-        JsonCompat.setPropertyNull(json, Constants.PROP_TRAITS); // prop
-        JsonCompat.setPropertyNull(json, Constants.PROP_MESSAGE); // prop
+        JsonCompat.setPropertyNull(json, Constants.PROP_TRAITS);
+        if (aaiNode.message != null) {
+            Object messageJson = writeMessageBase(aaiNode.message);
+            JsonCompat.setProperty(json, Constants.PROP_MESSAGE, messageJson);
+        } else {
+            JsonCompat.setPropertyNull(json, Constants.PROP_MESSAGE);
+        }
```

Comment:
So I think the way this should work is that the `message` child of the operation should be processed in a separate visit method called `visitMessage`.  The reason we set nulls for the child property values at this level is to force the ordering of the properties in the resulting JSON.  

So if there's a bug here, it's likely to be in the `visitMessage` method.

Reply:
Ok, I will revisit this and let you know my findings.